webapp/src/blocks.tsx

```diff
@@ -585,6 +585,35 @@ export class Editor extends toolboxeditor.ToolboxEditor {
                     return true
                 }
             });
+
+            const triggerEditorAction = (action: pxsim.SimulatorAction) => {
+                switch (action) {
+                    case "escape": {
+                        this.parent.setSimulatorFullScreen(false);
+                        return;
+                    }
+                    case "navigateregions" : {
+                        this.parent.showNavigateRegions();
+                        return
+                    }
+                }
+            }
+
+            const simulatorOrigins = [
+                window.location.origin,
+                // Simulator deployed origin.
+                "https://trg-microbit.userpxt.io"
+            ]
+            window.addEventListener("message", (e: MessageEvent) => {
```

Comment:
i'd rather this code go directly in app.tsx in `handleMessage`, which already handles the other messages that are sent from the sim

Reply:
2e84c145bc2f17c4a6353b54c2c4f6de8ec7761f

It made sense to move a bit more of the code into project view as the sim event and key event handling share code.

So far as I can tell that message handler doesn't check the origin of sim messages but I figure if that's going to change it should be a separate PR. Presumably it could do something like:

```typescript
const simulatorUrl = new URL(pxt.webConfig.simUrl, window.location.origin);
const simulatorOrigin = simulatorUrl.origin;
 window.addEventListener("message", ev => {
    if (ev.origin !== simulatorOrigin) {
        return;
    }
    // ...
```

?